File: 2661-smallest-missing-non-negative-integer-after-operations/smallest-missing-non-negative-integer-after-operations.java
class Solution {
    public int findSmallestInteger(int[] nums, int value) {
        int n = nums.length;

        int[] counts = new int[value + 1];
        for (int i = 0; i < n; i++) {
            counts[((nums[i] % value) + value) % value]++;
        }

        for (int i = 0; i < n; i++) {
            if (counts[i % value] == 0) return i;
            else {
                counts[i % value]--;
            } 
        }


        return n;
    }
}

File: 2661-smallest-missing-non-negative-integer-after-operations/smallest-missing-non-negative-integer-after-operations.py
class Solution:
    def findSmallestInteger(self, nums: List[int], value: int) -> int:
        n = len(nums)

        counts = [0] * (value + 1)
        for num in nums:
            counts[((num % value) + value) % value] += 1
        
        for i in range(n):
            if (counts[i % value] > 0):
                counts[i % value] -= 1
            else:
                return i

        return n